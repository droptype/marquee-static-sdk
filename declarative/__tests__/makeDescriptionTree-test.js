jest.autoMockOff()

describe('makeDescriptionTree', () => {
    const makeDescriptionTree           = require('../makeDescriptionTree')
    const React                         = require('react')
    const { HTMLView, Enumerate, Skip } = require('../Site')

    const r = React.createElement

    it('should parse the description', () => {
        const mock_description = (
            r(HTMLView, { props: {}, name: 'home', path: '/', component: () => r('div') },
                r(HTMLView, { props: {}, name: 'about', path: 'about', component: () => r('div') }),
                r(Enumerate, { items: [{slug: 'a'}, {slug: 'b'}, {slug: 'c'}] },
                    r(HTMLView, { props: s => s, name: 'story_detail', path: s => s.slug, linkKey: s => s, component: () => r('div') })
                )
            )
        )
        expect(
            makeDescriptionTree(mock_description)
        ).toMatchSnapshot()
    })

    it('should skip Skips', () => {
        const mock_description = (
            r(HTMLView, { props: {}, name: 'home', path: '/', component: () => r('div') },
                r(HTMLView, { props: {}, name: 'about', path: 'about', component: () => r('div') }),
                r(Skip, null,
                    r(Enumerate, { items: [{slug: 'a'}, {slug: 'b'}, {slug: 'c'}] },
                        r(HTMLView, { props: s => s, name: 'story_detail', path: s => s.slug, linkKey: s => s, component: () => r('div') })
                    ),
                )
            )
        )
        expect(
            makeDescriptionTree(mock_description)
        ).toMatchSnapshot()
    })

})