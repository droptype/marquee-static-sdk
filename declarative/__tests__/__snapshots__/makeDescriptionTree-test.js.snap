// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeDescriptionTree should parse the description 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "parent": [Circular],
      "props": Object {
        "component": [Function],
        "name": "about",
        "path": "about",
        "props": Object {},
      },
      "type": [Function],
    },
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "parent": [Circular],
          "props": Object {
            "component": [Function],
            "linkKey": [Function],
            "name": "story_detail",
            "path": [Function],
            "props": [Function],
          },
          "type": [Function],
        },
      ],
      "parent": [Circular],
      "props": Object {
        "items": Array [
          Object {
            "slug": "a",
          },
          Object {
            "slug": "b",
          },
          Object {
            "slug": "c",
          },
        ],
      },
      "type": [Function],
    },
  ],
  "parent": undefined,
  "props": Object {
    "component": [Function],
    "name": "home",
    "path": "/",
    "props": Object {},
  },
  "type": [Function],
}
`;
